test/run_all_tests.sh

```diff
@@ -114,14 +114,21 @@ else
   all_tests_result=999
 fi
 
-echo
 if [ 0 -eq ${all_tests_result} ]; then
   echo 'All tests have passed!'
-  gcovr --print-summary --sort-percentage --exclude-throw-branches --delete \
-    --object-directory . \
-    --root src \
-    --exclude .*_error.* \
-  && rm ./a.out *.gcno
+    gcovr --print-summary --sort-percentage --exclude-throw-branches --delete \
+      --object-directory . \
+      --root src \
+      --exclude .*_error.* \
+      --exclude test \
+      --coveralls coverage.json \
+      --txt && rm ./a.out *.gcno
+    if [ ! -f "coverage.json" ]; then
+      echo "Coverage report not produced";
+      all_tests_result=997
+    fi
```

Comment:
It would be nice to have matching levels of indentation. I don't care if it's two (2) spaces or four (4), just as long as they are the same.

Reply:
I agree! I don't see any difference in indentation in the VSCode editor, so presumably a tabs vs spaces issue? Will investigate...